sdk/servicebus/service-bus/src/connectionContext.ts

```diff
@@ -375,63 +425,76 @@ export namespace ConnectionContext {
         await connectionContext.managementClients[entityPath].close();
       }
 
-      // Calling onDetached on sender
-      if (!state.wasConnectionCloseCalled && state.numSenders) {
-        // We don't do recovery for the sender:
-        //   Because we don't want to keep the sender active all the time
-        //   and the "next" send call would bear the burden of creating the link.
-        // Call onDetached() on sender so that it can gracefully shutdown
-        //   by cleaning up the timers and closing the links.
-        // We don't call onDetached for sender after `refreshConnection()`
-        //   because any new send calls that potentially initialize links would also get affected if called later.
-        logger.verbose(
-          `[${connectionContext.connection.id}] connection.close() was not called from the sdk and there were ${state.numSenders} ` +
-            `senders. We should not reconnect.`
-        );
-        const detachCalls: Promise<void>[] = [];
-        for (const senderName of Object.keys(connectionContext.senders)) {
-          const sender = connectionContext.senders[senderName];
-          if (sender) {
-            logger.verbose(
-              "[%s] calling detached on sender '%s'.",
-              connectionContext.connection.id,
-              sender.name
-            );
-            detachCalls.push(
-              sender.onDetached().catch((err) => {
-                logger.logError(
-                  err,
-                  "[%s] An error occurred while calling onDetached() the sender '%s'",
-                  connectionContext.connection.id,
-                  sender.name
-                );
-              })
-            );
+      if (state.wasConnectionCloseCalled) {
+        // Do Nothing
+      } else {
+        // Calling onDetached on sender
+        if (state.numSenders) {
+          // We don't do recovery for the sender:
```

Comment:
`connectionContext` should also have the knowledge of the behavior of onDetached.

For instance, a few lines below.. for the streaming receiver, we need to call `onDetached()` on streaming receivers after the refreshConnection() since `onDetached()` would recover the streaming receivers and that would only be possible after the connection is refreshed.

So, here we need to have the knowledge of what's being done in onDetached to be able to call them in the right order.

Broadly speaking, as of now, there are only two buckets.
- [ sender, non-sessions-batching, sessions-(batching & streaming) ] 
- [ non-sessions-streaming ] 

If the behavior of "sessions" ever changes, and we consider recovering them upon a disconnect, we'd move sessions-streaming to the second bucket.

> Because of that I think having this documentation in Sender makes more sense.

I'd say both the places instead.